### DISCUSSION TITLE
Prevent 8bit Grayscale Image on Image Creation

### DISCUSSION THREAD

**Discussion by mpoelzl** [dlemstra/Magick.NET#1186] [Help] [answered]
Hello!

magickImage.Write(new FileInfo(fullFileName), MagickFormat.Jpg);

I am trying to prevent Magick.NET from creating 8bit grayscale images when there is no colour in the image data (this seems to be an automatic thing) i.e. I would always want the images to be saved in 24bit colour even if the contained data is grayscale. I have tried to set various things to force this but haven't found a way to achieve it.

Thanks for your help!

Mike

**Comment by dlemstra** [comment]
Can you share a sample input image?

**Reply by mpoelzl** [reply]
The image is created from raw data like this:

        var readSettings = new MagickReadSettings
        {
            Width = stride / 3,
            Height = totalImageHeight,
            Format = MagickFormat.Rgb
        };

        return new MagickImage(imageBuffer, readSettings);

**Comment by dlemstra** [comment]
I cannot use that example to reproduce your question. Please prove a small example that can be used to reproduce this question.

**Reply by mpoelzl** [reply]
using ImageMagick;

byte[] data = new byte[30000];

// If commented out an 8bit image will be saved. Otherwise 24 bit 
data[0] = 1;

var readSettings = new MagickReadSettings
{
    Width = 100,
    Height = 100,
    Format = MagickFormat.Rgb
};

new MagickImage(data, readSettings).Write(new FileInfo("test.jpg"), MagickFormat.Jpg);

**Reply by mpoelzl** [reply]
Just to repeat the question. I would like to force Magick.Net to always save a 24bit image, even if the image is grayscale. The references assembly is: Magick.NET-Q8-AnyCPU

**Comment by dlemstra** [comment, marked as answer]
You can force 24 bit images when you set the following settings right before you write the image.

```C#
image.ColorSpace = ColorSpace.sRGB;
image.Settings.ColorType = ColorType.TrueColor;
```

**Reply by mpoelzl** [reply]
Thank you! I was missing the ColorType settings...
